controller/repliesController.js

```diff
@@ -159,104 +162,73 @@ const getReplyVotes = async (req, res, next) => {
 
     const reply = await Replies.findById(replyId);
 
-    // Check to see if the reply exists.
+// Check to see if the reply exists.
     if (!reply) {
       return next(
         new CustomError(
           404,
-          `Reply with the ID ${replyId} does not exist or has been deleted`
+          'Reply with the ID ${replyId} does not exist or has been deleted'
         )
       );
     }
 
-    // A list of all the votes
-    const votes = [];
-
-    if (!voteType) {
-      // Add all votes
-      votes.push(...reply.upVotes);
-      votes.push(...reply.downVotes);
-    } else {
-      if (voteType === "upvote") {
-        // Add upvotes only
-        votes.push(...reply.upVotes);
-      }
 
-      if (voteType === "downvote") {
-        // Add downvotes only
-        votes.push(...reply.downVotes);
-      }
-    }
-
-    // The data object to be returned in the response
-    const data = {
-      replyId,
-      commentId,
-      votes,
-    };
-
-    return responseHandler(res, 200, data, "OK");
-  } catch (error) {
+        //if user exists in downvotes array
+        if (comment.downVotes.includes(ownerId)) {
+          //get index of user in downvotes array
+          const voterIndex = comment.reply.downVotes.indexOf(ownerId);
+          //if index exists
+          if (voterIndex > -1) {
+            //delete that index
+            comment.reply.downVotes.splice(voterIndex, 1);
+          }
+        }
+
+        //same as above for upvotes
+        if (comment.reply.upVotes.includes(ownerId)) {
+          const voterIdx = comment.reply.upVotes.indexOf(ownerId);
+          if (voterIdx > -1) {
+            comment.reply.upVotes.splice(voterIdx, 1);
+          }
+        } else {
+          // add user to the top of the upvotes array
+          comment.reply.upVotes.unshift(ownerId);
+        }
+
+        //save the comment vote
+        comment.save();
+
+        //get total number of elements in array
+        const totalUpVotes = comment.reply.upVotes.length;
+        const totalDownVotes = comment.reply.downVotes.length;
+
+        //get total number of votes
+        const totalVotes = totalUpVotes + totalDownVotes;
+
+        const data = {
+          commentId: comment._id,
+          numOfVotes: totalVotes,
+          numOfUpVotes: totalUpVotes,
+          numOfDownVotes: totalDownVotes,
+        };
```

Comment:
Please add the replyId too. Also you already have the commentId from the params. It's not necessary for you to fetch the _id of the comment :
```suggestion
        const data = {
          replyId,
          commentId,
          numOfVotes: totalVotes,
          numOfUpVotes: totalUpVotes,
          numOfDownVotes: totalDownVotes,
        };
```